GiveNGo/GiveNGo
# Setup Jest and Typescript

## Description:
- Initial setup and tests for our React-Native application to incorporate Jest and Enzyme tests for React components. The inclusion of Jest has added additional testing scripts within the **package.json** file, as well as new configuration settings for **tsconfig.json**

## Changes I Made:
- Added **/build** directory to our **.gitignore** file 
-  Created initial React-Native unit test within the **__tests__** testing directory
- Setup the **jest.config.json** file including the testing environment
- Provided additional compiler options within the **tsconfig.json** file
- Included new scripts to test our application, and allow **Jest to enter watch mode** while creating additional tests

<img width="738" alt="Screen Shot 2020-07-21 at 11 22 28 AM" src="https://user-images.githubusercontent.com/55554784/88091938-923e6a00-cb44-11ea-8013-6d0eb71b8102.png">


## How to Test:
- `npm install`
- `npm run test`
- This script should return a successful test on our` <App /> `component 
- Additionally, run `tsc` in your terminal to ensure a build file is generated 

<img width="427" alt="Screen Shot 2020-07-21 at 11 12 41 AM" src="https://user-images.githubusercontent.com/55554784/88090893-198ade00-cb43-11ea-8d74-d1d2cfc3e67b.png">
